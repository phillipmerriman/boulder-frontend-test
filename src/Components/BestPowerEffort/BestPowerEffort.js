import React from 'react';

const BestPowerEffort = (props) => {
  const { twentyMinute, oneMinute } = props;

  return (
    <>
      <h2 data-testid='bestOneMinute'>Best 1 Minute Power Effort: {oneMinute}</h2>
      <h2 data-testid='bestTwentyMinute'>Best 20 Minute Power Effort: {twentyMinute}</h2>
    </>
  );
};

export default BestPowerEffort;
