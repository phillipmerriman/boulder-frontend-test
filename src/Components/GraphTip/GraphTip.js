import React from 'react';

// just a little component explaining the X and Y axis for the user
const GraphTip = () => {
  return <h3 data-testid='axisInfo'>X-Axis: Milliseconds | Y-Axis: Power Output</h3>;
};

export default GraphTip;
