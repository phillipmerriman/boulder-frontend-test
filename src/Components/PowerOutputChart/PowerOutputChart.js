import React from "react";
import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  ResponsiveContainer,
} from "recharts";

import GraphTip from "../GraphTip/GraphTip";

const PowerOutputChart = (props) => {
  const { msAndPower } = props;

  return (
    <>
      <ResponsiveContainer width="100%" aspect={3}>
        <LineChart
          width={500}
          height={300}
          data={msAndPower}
          margin={{
            top: 5,
            right: 30,
            left: 20,
            bottom: 5,
          }}
        >
          <CartesianGrid />
          <XAxis dataKey="millisecondOffset" />
          <YAxis />
          <Tooltip />
          <Line
            type="monotone"
            dataKey="power"
            stroke="#8884d8"
            activeDot={{ r: 8 }}
          />
        </LineChart>
      </ResponsiveContainer>
      <GraphTip />
    </>
  );
};

export default PowerOutputChart;
