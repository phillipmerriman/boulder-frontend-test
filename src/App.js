import React from 'react';

import { calculatePower } from './Utils/calculatePower';
import BestPowerEffort from './Components/BestPowerEffort/BestPowerEffort';
import PowerOutputChart from './Components/PowerOutputChart/PowerOutputChart';
import './App.css';
import data from './workout-data.json';

function App() {
  // create array of objects out of workout-data.json, grabbing only relevant data for the graph
  const msAndPower = data.samples.map((sample) => {
    return {
      millisecondOffset: sample.millisecondOffset,
      power: sample.values.power,
    };
  });

  // create array of just the power output for calculating the best power output over time
  const power = msAndPower.map((object) => object.power);

  // get best twenty minute power output
  const twentyMinute = calculatePower(power, 1200);
  // get best one minute power output
  const oneMinute = calculatePower(power, 60);

  return (
    <div className='App'>
      <header className='App-header'>
        <BestPowerEffort twentyMinute={twentyMinute} oneMinute={oneMinute} />
        <PowerOutputChart msAndPower={msAndPower} />
      </header>
    </div>
  );
}

export default App;
  