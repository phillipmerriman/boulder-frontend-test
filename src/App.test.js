import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

import App from './App';
import BestPowerEffort from './Components/BestPowerEffort/BestPowerEffort';

import PowerOutputChart from './Components/PowerOutputChart/PowerOutputChart';
import GraphTip from './Components/GraphTip/GraphTip';

describe('Parent component - App.js', () => {

    afterEach(cleanup);
  
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
    });
    
    it('renders best 1 minute power effort correctly', () => {
        render(<BestPowerEffort oneMinute={300} />);
        expect(screen.getByTestId('bestOneMinute')).toHaveTextContent(300);
    });
  
    it('renders best 20 minute power effort correctly', () => {
        render(<BestPowerEffort twentyMinute={200} />);
        expect(screen.getByTestId('bestTwentyMinute')).toHaveTextContent(200);
    });

    it('renders PowerOutputChart correctly', () => {
        render(<PowerOutputChart />);
        expect(screen.getByTestId('powerOutputChart')).toBeInTheDocument();
    });
    
    it('renders axis info correctly', () => {
        render(<GraphTip />);
        expect(screen.getByTestId('axisInfo')).toHaveTextContent('X-Axis: Milliseconds | Y-Axis: Power Output');
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
  
});
