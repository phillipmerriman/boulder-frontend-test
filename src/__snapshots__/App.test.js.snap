// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parent component - App.js matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h2
      data-testid="bestOneMinute"
    >
      Best 1 Minute Power Effort: 
      376.31666666666666
    </h2>
    <h2
      data-testid="bestTwentyMinute"
    >
      Best 20 Minute Power Effort: 
      231.95666666666668
    </h2>
    <div
      data-testid="powerOutputChart"
    >
      <div
        className="recharts-responsive-container"
        style={
          Object {
            "height": "100%",
            "maxHeight": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "width": "100%",
          }
        }
      />
      <h3
        data-testid="axisInfo"
      >
        X-Axis: Milliseconds | Y-Axis: Power Output
      </h3>
    </div>
  </header>
</div>
`;
